src/pybind/mgr/diskprediction/module.py

```diff
@@ -0,0 +1,346 @@
+"""
+A diskprediction module
+"""
+from __future__ import absolute_import
+
+from datetime import datetime
+import json
+from mgr_module import MgrModule
+from threading import Event
+
+from .common import DP_MGR_STAT_ENABLED, DP_MGR_STAT_DISABLED
+from .task import MetricsTask, PredictionTask, SmartTask
+
+
+DP_TASK = [MetricsTask, PredictionTask, SmartTask]
+
+
+class Module(MgrModule):
+
+    config = dict()
```

Comment:
No need for an additional copy of all the config -- just use get_config when you need something.